src/views/RouteArviointiasteikot.vue

```diff
@@ -1,36 +1,136 @@
+        <ep-button
+          class="ml-2"
+          v-if="isEditing"
+          variant="primary"
+          @click="saveArviointiAsteikko()"
+          :show-spinner="isSaving">
+          <slot name="tallenna">{{ $t('tallenna') }}</slot>
+        </ep-button>
+      </div>
+    </div>
+    <div class="taso-wrapper">
+      <template v-if="arviointiasteikot">
+        <div class="asteikko mt-4" v-for="(asteikko, idx) in arviointiasteikot" :key="asteikko.id">
+          <span class="text-nowrap">
+            <h3>{{$t('arviointiasteikko') + ' ' + (idx+1)}}</h3>
+            <ep-spinner v-if="!asteikko.osaamistasot" />
+            <div v-else>
+              <div
+                v-for="(taso, index) in asteikko.osaamistasot"
+                :key="taso.id"
+                class="taso py-2"
+                :class="{ 'pl-3': !isEditing, 'is-editing': isEditing }">
+                <template v-if="!isEditing">
+                  {{ $kaanna(taso.otsikko) }}
+                </template>
+                <b-form-group
+                  :label="$t('osaamistaso') + ' ' + (index + 1)"
+                  v-else>
+                  <ep-input
+                    :name="$t('osaamistaso')"
+                    v-model="taso.otsikko"
+                    :is-editing="true"
+                    @input="setOsaamistasoOtsikko"/>
+                </b-form-group>
+              </div>
+            </div>
+          </span>
+        </div>
+      </template>
+      <ep-spinner v-else />
     </div>
   </div>
 </template>
 
 <script lang="ts">
 import { Prop, Component, Vue } from 'vue-property-decorator';
+
 import EpMainView from '@shared/components/EpMainView/EpMainView.vue';
 import EpIcon from '@shared/components/EpIcon/EpIcon.vue';
 import EpJulkiLista from '@shared/components/EpJulkiLista/EpJulkiLista.vue';
+import EpButton from '@shared/components/EpButton/EpButton.vue';
+import EpSpinner from '@shared/components/EpSpinner/EpSpinner.vue';
+import EpInput from '@shared/components/forms/EpInput.vue';
+
 import { ArviointiStore } from '@/stores/ArviointiStore';
 
+import { ArviointiAsteikkoDto, OsaamistasoDto } from '@shared/api/eperusteet';
+
 @Component({
   components: {
     EpMainView,
     EpIcon,
     EpJulkiLista,
+    EpButton,
+    EpSpinner,
+    EpInput,
   },
 })
 export default class RouteArviointiasteikot extends Vue {
   @Prop({ required: true })
   arviointiStore!: ArviointiStore;
 
-  mounted() {
-    this.arviointiStore.fetchArviointiasteikot();
+  private isEditing = false;
+  private isSaving = false;
+  private arviointiAsteikkoData: ArviointiAsteikkoDto[] | undefined = undefined;
+
+  async mounted() {
+    await this.arviointiStore.fetchArviointiasteikot();
     this.arviointiStore.fetchGeneeriset();
+    this.arviointiAsteikkoData = this.arviointiStore.arviointiasteikot.value as ArviointiAsteikkoDto[];
+  }
+
+  toggleEdit() {
+    this.isEditing = !this.isEditing;
+  }
+
+  setOsaamistasoOtsikko(data: OsaamistasoDto) {
```

Comment:
tämä on turha, jos käyttää suoraan arviointiAsteikkoData:aa v-for:ssa. 

Jotta consoli ei ala valittamaan, pitää private arviointiAsteikkoData: 
ArviointiAsteikkoDto[] | undefined = undefined muuttaa -> private arviointiAsteikkoData: ArviointiAsteikkoDto[] | null= null;